### DISCUSSION TITLE
SUpport for Podman

### DISCUSSION THREAD

**Discussion by HilkopterBob** [louislam/dockge#535] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Support Podman as a drop-in replacement for docker and docker compose.
With an option inside the settings to change the container engine from docker (default) to podman & podman compose

### ✔️ Solution

[Podman-Compose](https://github.com/containers/podman-compose) makes it possible to use docker compose commands and yaml files to build and run containers for Podman.  I guess the most work would need to happen inside the dockge-module that actually calls/runs the docker shell commands.

### 📝 Additional Context

Podman is a container engine like docker, with a docker compatibility layer, that utilizes a root and daemon less architecture to minimize the attack surface for eg. privilege escalation.

Podman compose is a comunity made script that enables the use of docker compose like commands and yaml fies.

**Comment by ParaplegicRacehorse** [comment]
I spotted https://github.com/louislam/dockge/pull/515 in waiting pull-requests. It just updates the README, which suggests the capability is already baked into dockge; sort of.

I think it would be great to just have it find my local podman.socket as an agent without having to spin up another container.

**Comment by karoltheguy** [comment]
Would be great if Rootless Podman and Quadlets were supported.
